### PR TITLE
OOP school library: preserve data

### PR DESCRIPTION
**Pull request by VaheStepanyan100**
- On exit from the app saved data (books, people, and rentals) into JSON files.
- On startup of the app loaded all data from files. handled the case when one or more of the files are missing.

### CODE REVIEW

**Comment by prolajumokeoni** [comment]
Hi @VaheStepanyan100 and @AbdelazizSoliman, please fix the Rubocop linter error 👍🏿

**Comment by VaheStepanyan100** [comment]
Hi @prolajumokeoni  Thanks for the heads up! The Rubocop error has been fixed and the codebase is now clean

File: user_interactions.rb [hunk 1]
```diff
@@ -61,4 +65,17 @@ def list_numbers
...
+    app_arrays = %w[people books rentals]
+
+    app_arrays.each do |file|
+      data = @app.instance_variable_get("@#{file}")
+      json_data = data.map(&:to_hash)
+      create_file(file, json_data)
+    end
+
+    puts 'Thank you for using this app!'
+  end
```
Comment (DammyShittu) [inline user_interactions.rb:80]: - [ ] There seems to be an error somewhere in your app. When I added books, people, and rentals and quit the app, it did not record the new data in the json storage files. Kindly fix that.

- [ ] When I deleted the existing json storage files and created new people, books, and rentals, and quit the app, the files are not created and the data is not saved. The app mostly runs like the previous basic UI project, no new data is persisted in any scenario. Kindly fix these issues.

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @VaheStepanyan100 and @AbdelazizSoliman,

## Highlights
✔️ Passing linters
✔️ Working program

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by joskalenda — APPROVED** [review]
| STATUS: APPROVED ✅ |
|------------------------|
| Hi @VaheStepanyan100 , 🥰|
| Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: Congratulations! 🎉  |
|  |

### Highlights
✅ ... Greate and clean code
✅ ... Readme file

---------
### Optional suggestions

> _Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

### Cheers and Happy coding!👏👏👏
-------
> Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me(@joskalenda) in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**